rust/common/error/macros.rs

```diff
@@ -62,6 +62,19 @@ macro_rules! error_messages {
                     _ => unreachable!(),
                 }
             }
+
+            const fn name(&self) -> &'static str {
+                match self {$(
+                    Self::$error_name(..) => std::stringify!($error_name),
+                )*}
+            }
+
+            fn format_payload<'a, 'b>(&'a self, debug_struct: &'a mut std::fmt::DebugStruct<'a, 'b>) -> &mut std::fmt::DebugStruct<'a, 'b>
+                where 'b: 'a
+            {
+                $(error_messages!(@payload self, debug_struct, $error_name $(, $inner)*);)*
+                unreachable!()
+            }
```

Comment:
Does this helper need to exist? This is specific to `fmt::Debug` and could be inlined.

Reply:
It doesn't need to exist, and has been removed.